material/material-theme-text-field.html

```diff
@@ -0,0 +1,121 @@
+<dom-module id="material-theme-text-field" theme-for="vaadin-text-field">
+  <template>
+    <style>
+      :host {
+        position: relative;
+        width: 160px;
+        padding-top: 16px;
+        margin-bottom: 8px;
+        color: rgba(0, 0, 0, 0.87);
+        -webkit-font-smoothing: antialiased;
+        -moz-osx-font-smoothing: grayscale;
+
+        /* Fix Chrome’s dancing labels */
+        contain: content;
+
+        /*
+        -webkit-transition needed for Safari right now, but it’s fixed in
+        Safari Technology Preview, so we can expect it to fix itself in the
+        near future (Safari 10.2 most probably).
+        */
+        -webkit-transition: background-size 0.25s;
+        transition: background-size 0.25s;
+
+        background-size: 0% 2px, 100% 2px;
+        background-image: linear-gradient(#3F51B5, #3F51B5), linear-gradient(rgba(0, 0, 0, 0.12) 1px, transparent 1px);
+        background-repeat: no-repeat;
+        background-position: 50% 100%;
+      }
+
+      [part~="input"] {
+        outline: none;
+        margin: 0;
+        border: 0;
+        border-radius: 0;
+        padding: 8px 0 10px;
+        width: 100%;
+        height: calc(16px + 8px + 10px);
+        font-family: Roboto, sans-serif;
+        font-size: 16px;
+        color: inherit;
+        background-color: transparent;
+
+        /* Disable default invalid style in Firefox */
+        box-shadow: none;
+      }
+
+      [part~="input"]::-webkit-input-placeholder {
+        color: rgba(0, 0, 0, 0.38);
+      }
+
+      [part~="input"]::-moz-placeholder {
+        color: rgba(0, 0, 0, 0.38);
+      }
+
+      [part~="input"]::placeholder {
+        color: rgba(0, 0, 0, 0.38);
+      }
+
+      [part~="label"] {
+        display: block;
+        font-family: Roboto, sans-serif;
+        font-size: 16px;
+        line-height: 1;
+        white-space: nowrap;
+        overflow: hidden;
+        text-overflow: ellipsis;
+        color: rgba(0, 0, 0, 0.38);
+        transform-origin: 0 50%;
+        transform: scale(0.75);
+        width: 133%;
+
+        -webkit-transition: transform 0.25s;
+        transition: transform 0.25s;
+      }
+
+      :host([focused]) {
+        background-size: 100% 2px, 100% 2px;
+      }
+
+      :host([focused]) [part~="label"] {
+        color: #3F51B5;
+      }
+
+      :host([invalid]) {
+        background-image: linear-gradient(#dd2c00, #dd2c00);
+        background-size: 100% 2px;
+      }
+
+      :host([invalid]) [part~="label"] {
+        color: #dd2c00;
+      }
+
+      :host([disabled]) {
+        color: rgba(0, 0, 0, 0.38);
+        background-image: linear-gradient(90deg, rgba(0, 0, 0, 0.24) 0, rgba(0, 0, 0, 0) 2px);
```

Comment:
Underline color is too low contrast according to the specs. Should be 0.42 instead.